server/player/player.go

```diff
@@ -1909,7 +1913,82 @@ func (p *Player) close() {
 		p.World().RemoveEntity(p)
 	} else {
 		s.CloseConnection()
+		p.Save()
+	}
+}
+
+// Load reads the player data from the provider. It uses the default values if the provider
+// returns false.
+func (p *Player) Load() {
+	data, ok := p.provider.Load(p.UUID())
+	if !ok {
+		return
+	}
+
+	//p.pos.Store(data.Position)
+	p.Teleport(data.Position.Sub(mgl64.Vec3{0.5, 0, 0.5}))
+	p.Rotate(data.Yaw-p.yaw.Load(), data.Pitch-p.pitch.Load())
```

Comment:
Perhaps this yaw and pitch should be sent in the start game packet, including the position, so you don't have to teleport the player?

Reply:
Thanks, that does work!